### ISSUE TITLE
After deleting an image from a contenttype, the image data is still visible 

### ISSUE THREAD

**Issue by milosa** [bolt/core#1911]
I have a contenttype (singleton, viewless), with an image contenttype:

```
        homepage_picture:
            label: Homepage afbeelding
            type: image
            alt: true
```

In the template, I try to check if the image exist with:

```
{% if homepage.homepage_picture %}do something{% endif %}
```

When I start with a new content type, and don't select an image, 'do something' will not show.  (as expected)
When I select an image (from the library), it (the text) will show. (as expected)
But when I then remove the image (and save the content type, and clear cache), it (the text) will stay visible.

Also, when I do `{{ dump(homepage.homepage_picture|media) }}`, it will keep showing the data for the image I just removed. 
When I try to display the image (after I removed it) with `<img src="{{ homepage.homepage_picture }}" alt="{{ homepage.homepage_picture.alt }}">`, just `<img src="/files/" alt="">` will be outputted.

Details
-------

| Question                      | Answer
|----------------- | ------
| Relevant Bolt Version | 4.0
| Install type                   |  composer
| PHP version                 | 7.4.10 |
| Web server                  | Apache / docker |

**Comment by bobdenotter** [comment]
> But when I then remove the image

Do you mean removing the image itself from the `public/files` folder? Or clearing the Field when editing the page?

**Comment by milosa** [comment]
The latter, using the remove/delete button under the upload field. Then saving the content type item in the backend.

**Comment by bobdenotter** [comment]
This is mostly fixed in #1925: It fixes it returning `/files/`, and the `|media` thing. 

What isn't fixed is: 

```twig
{% if homepage.homepage_picture %}do something{% endif %}
```

Because the Field is still an entity object, even if it's empty. I'm not sure if that's something we can fix. It'd be better if you checked it like this, which will behave as expected: 

```twig
{% if homepage.homepage_picture is not empty %}do something{% endif %}
```

**Comment by milosa** [comment]
@bobdenotter, The `|media` thing still isn't fixed. It still returns the data for the image that has been removed from the item.

But I also found some other weird things.

When I have a contenttype with a collection/set, with an image in it. The issue still exists. 
The fieldtype looks something like this:

```
homepage_sections:
            type: collection
            label: Homepage sectie
            fields:
                section:
                    type: set
                    fields:
                        section_body:
                            type: redactor
                            label: Sectie inhoud
                        section_image:
                            label: Afbeelding
                            type: image
                            alt: true
```

When I don't upload or select an image to the section, the `is empty` check still doesn't work properly.
This code shows `<h1>hallo</h1>` as output for all sections. Even when no image is present. The |media dump shows `null` in this case.

```
    {% for section in homepage.homepage_sections %}
        {% if section.section_image is not empty %}<h1>hallo</h1>{% endif %}
        {{ dump(section.section_image|media) }}
```
And when I do add an image, and then remove it again, it keeps showing the `<h1>hallo</h1>` and it also keeps showing the full media data of the removed image. The image isn't shown when trying to display it with `thumbnail()` or something.

So, long story short, images don't work properly with collections/sets.

**Comment by bobdenotter** [comment]
Weird. I'm not able to reproduce this yet.. I'll re-open the issue, and we'll see if others bump into it as well.
